core/store/src/flat/storage.rs

```diff
@@ -255,13 +255,20 @@ impl FlatStorage {
         let mut deltas = HashMap::new();
         for delta_metadata in deltas_metadata {
             let block_hash = delta_metadata.block.hash;
-            let changes: CachedFlatStateChanges =
+            let changes: CachedFlatStateChanges = if delta_metadata
+                .prev_block_with_changes
+                .is_none()
```

Comment:
It's a bit confusing to me that if the prev block with changes is none then read the delta changes and if it's some then we'll default it. Shouldn't it be the other way around? 

Sanity check: Does this work for end of epoch where there may be no receipts in the block but the rewards are still distributed?